packages/dds/tree/src/test/simple-tree/api/schemaFactory.examples.spec.ts

```diff
@@ -127,4 +128,16 @@ describe("Class based end to end example", () => {
 			}),
 		);
 	});
+
+	it("customized narrowing", () => {
+		class Specific extends schema.object("Specific", {
+			s: customizeSchemaTyping(schema.string).simplified<"foo" | "bar">(),
```

Comment:
This shows customizeSchemaTyping being used to implement a previously requested feature of narrowing leaf types.

It can also be used for branding of lead types, which would be even safer from a soundness perspective.

Reply:
This looks cool